### PR TITLE
Upgraded to Apollo Client v3

### PR DESCRIPTION
**Pull request by Vultraz**
Getting a head start on this. Apollo Client v3 is still in the release candidate stage, so this could either be merged now (easier to work with, this gets testing), or we wait until 3.0 proper.

This doesn't touch `@keystonejs/apollo-helpers`, which I think should be removed.

### CODE REVIEW

File: packages/app-admin-ui/client/apolloClient.js [hunk 1]
```diff
@@ -1,21 +1,10 @@
...
-import { onError } from 'apollo-link-error';
-import { InMemoryCache, IntrospectionFragmentMatcher } from 'apollo-cache-inmemory';
+import { onError } from '@apollo/link-error';
+import { InMemoryCache } from '@apollo/client/cache';
 
 import fetch from 'cross-fetch';
 
-// This shouldn't be necessary, but it silences an annoying error from Apollo
-// Client: https://github.com/apollographql/apollo-client/issues/3397#issuecomment-421433032
-const fragmentMatcher = new IntrospectionFragmentMatcher({
```
Comment (Vultraz) [inline packages/app-admin-ui/client/apolloClient.js:11]: Need to confirm if the error this was meant to fix is gone on 3.0, but `IntrospectionFragmentMatcher` has been removed. We're supposed to use the `possibleTypes` InMemoryCache config option instead. Will need to look into that if the error is still around.

Comment (jordanoverbye) [reply packages/app-admin-ui/client/apolloClient.js:11]: Yeah I was confused about this too... Do you know which queries were causing these errors? I was running these changes locally, and I couldn't reproduce any errors. 

I think the below code is the same behaviour as the deleted code - but Im not sure if we need to keep this.

```javascript
new InMemoryCache({
  possibleTypes: {
    __schema: [],
  }
})
```

Comment (Vultraz) [reply packages/app-admin-ui/client/apolloClient.js:11]: I *think* had something to do with the Unsplash field? @jesstelford would know more, I think.

**Comment by timleslie** [comment]
@jordanoverbye I know you've been doing some Apollo 3 stuff lately. Could you take a look at this branch and see if you can help @Vultraz to track down the test failures here?

**Comment by Vultraz** [comment]
It's just the column-spec test. It seems the behavior when selecting a non-cached filter is slightly different in v3 (you enter a loading state), so the column selection dialog doesn't always remain open as the tests test the different filters. I was trying to find a way to open it if it's closed, but Cypress doesn't seem to have a way to do that simply.

**Comment by timleslie** [comment]
If it turns out that this is too difficult to solve then there is always the option to remove the test. This is a case where the tests are providing negative value (e.g. costing us time while telling us very little about the correctness of our code). If the answer is to not run the test then I'd be happy with that 👍

**Comment by Vultraz** [comment]
@timleslie green 👍

File: demo-projects/meetup/site/components/auth/forgotPassword.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 /** @jsx jsx */
 
 import { useState, useEffect } from 'react';
-import { useMutation } from '@apollo/react-hooks';
+import { useMutation } from '@apollo/client';
 import Router from 'next/router';
 import { jsx } from '@emotion/core';
-import gql from 'graphql-tag';
+import { gql } from '@apollo/client';
```
Comment (timleslie) [inline demo-projects/meetup/site/components/auth/forgotPassword.js:7]: Let's merge this with the other import.

File: test-projects/basic/cypress/integration/add-edit-delete-data.js [hunk 3]
```diff
@@ -54,6 +54,7 @@ describe('Adding data', () => {
     cy.get('#ks-input-author').click({ force: true });
     cy.get('#ks-input-author').type('John Doe{downarrow}{enter}', {
       force: true,
+      delay: 100,
```
Comment (timleslie) [inline test-projects/basic/cypress/integration/add-edit-delete-data.js:57]: Is this change still required for the tests to work?

File: demo-projects/blog/package.json [hunk 4]
```diff
@@ -33,14 +32,11 @@
...
-    "apollo-cache-inmemory": "^1.6.6",
-    "apollo-client": "^2.6.10",
-    "apollo-upload-client": "^13.0.0",
+    "apollo-upload-client": "^14.1.0",
     "cross-env": "^7.0.0",
     "date-fns": "^2.14.0",
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
-    "graphql-tag": "^2.10.3",
     "isomorphic-unfetch": "^3.0.0",
```
Comment (jordanoverbye) [inline demo-projects/blog/package.json:40]: I think we should add `graphql` as a dependency here as per the [apollo docs](https://www.apollographql.com/docs/react/get-started/). It looks like its just missing from the demo projects

File: demo-projects/meetup/package.json [hunk 5]
```diff
@@ -39,7 +36,6 @@
     "express": "^4.17.1",
     "facepaint": "^1.2.1",
     "get-contrast": "^2.0.0",
```
Comment (jordanoverbye) [inline demo-projects/meetup/package.json:38]: Same as above

File: test-projects/basic/cypress/integration/add-edit-delete-data.js (hunk 3 above)
Comment (Vultraz) [reply test-projects/basic/cypress/integration/add-edit-delete-data.js:57]: Yes, it now needed a short delay after typing for the results to appear in the dropdown, else it would false.

File: demo-projects/meetup/site/components/auth/forgotPassword.js (hunk 2 above)
Comment (Vultraz) [reply demo-projects/meetup/site/components/auth/forgotPassword.js:7]: Done

File: demo-projects/blog/package.json (hunk 4 above)
Comment (Vultraz) [reply demo-projects/blog/package.json:41]: Done

File: demo-projects/meetup/package.json (hunk 5 above)
Comment (Vultraz) [reply demo-projects/meetup/package.json:38]: Done

File: demo-projects/blog/package.json (hunk 4 above)
Comment (jordanoverbye) [reply demo-projects/blog/package.json:41]: Thanks @Vultraz !
